crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -611,6 +614,65 @@ impl Mesh {
         }
     }
 
+    /// Calculates the [`Mesh::ATTRIBUTE_NORMAL`] of an indexed mesh, smoothing normals for shared
+    /// vertices.
+    ///
+    /// # Panics
+    /// Panics if [`Mesh::ATTRIBUTE_POSITION`] is not of type `float3`.
+    /// Panics if the mesh has any other topology than [`PrimitiveTopology::TriangleList`].
+    /// Panics if the mesh does not have indices defined.
+    ///
+    /// FIXME: This should handle more cases since this is called as a part of gltf
+    /// mesh loading where we can't really blame users for loading meshes that might
+    /// not conform to the limitations here!
+    pub fn compute_smooth_normals(&mut self) {
+        assert!(
+            matches!(self.primitive_topology, PrimitiveTopology::TriangleList),
+            "`compute_smooth_normals` can only work on `TriangleList`s"
+        );
+        assert!(
+            self.indices().is_some(),
+            "`compute_smooth_normals` can only work on indexed meshes"
+        );
+
+        let positions = self
+            .attribute(Mesh::ATTRIBUTE_POSITION)
+            .unwrap()
+            .as_float3()
+            .expect("`Mesh::ATTRIBUTE_POSITION` vertex attributes should be of type `float3`");
+
+        let mut count: usize = 0;
+        let mut corners = [0_usize; 3];
+        let mut normals = vec![[0.0f32; 3]; positions.len()];
+        let mut adjacency_counts = vec![0_usize; positions.len()];
+        for i in self.indices().unwrap().iter() {
+            corners[count % 3] = i;
+            count += 1;
```

Comment:
This `corner` + `count` stuff seem to be identical as a `chunk_exact(3)` behaviour, which we do use in `compute_flat_normals`, we should probably pick a single method between the two ways.
I prefer using `chunk_exact` as is it more readable than this manual way of doing it